### PR TITLE
Feat: keep funding reserve on exchange

### PR DESCRIPTION
**Pull request by sebastienverreault**

### CODE REVIEW

File: dealer/src/OkexPerpetualSwapStrategy.ts [hunk 1]
```diff
@@ -525,75 +525,109 @@ export class OkexPerpetualSwapStrategy implements HedgingStrategy {
...
-              await database.externalTransfers.insert(extTransferRecord)
-              return { ok: false, error: withdrawalResult.error }
+            if (fundingAccountBalanceResult.ok) {
+              newTransferSizeInBtc = Math.max(
+                0,
+                fundingAccountBalanceResult.value.btcFreeBalance -
+                  hedgingBounds.MINIMUM_FUNDING_BALANCE_BTC,
+              )
+            } else {
+              this.logger.error(
```
Comment (dolcalmi) [inline dealer/src/OkexPerpetualSwapStrategy.ts:544]: not sure to understand this.. if `this.getFundingAccountBalance()` fail then we continue trying to withdraw?

also, I prefer the other style that you use for ok returns, something like line 584

```
if (!fundingAccountBalanceResult.ok) {
  this.logger.error(
    { fundingAccountBalanceResult },
    "getFundingAccountBalance() error: {fundingAccountBalanceResult}",
  )
  return { ... }
}

newTransferSizeInBtc = Math.max(
  0,
  fundingAccountBalanceResult.value.btcFreeBalance -
    hedgingBounds.MINIMUM_FUNDING_BALANCE_BTC,
)
```

File: dealer/src/OkexPerpetualSwapStrategy.ts [hunk 2]
```diff
@@ -525,75 +525,109 @@ export class OkexPerpetualSwapStrategy implements HedgingStrategy {
...
-            }
-
-            if (!withdrawalResult.ok) {
-              await database.externalTransfers.insert(extTransferRecord)
-              return { ok: false, error: withdrawalResult.error }
+            if (fundingAccountBalanceResult.ok) {
+              newTransferSizeInBtc = Math.max(
+                0,
+                fundingAccountBalanceResult.value.btcFreeBalance -
+                  hedgingBounds.MINIMUM_FUNDING_BALANCE_BTC,
```
Comment (dolcalmi) [inline dealer/src/OkexPerpetualSwapStrategy.ts:541]: any reason to just not do a comparison?  `fundingAccountBalanceResult.value.btcFreeBalance > hedgingBounds.MINIMUM_FUNDING_BALANCE_BTC`

File: dealer/src/OkexPerpetualSwapStrategy.ts [hunk 3]
```diff
@@ -525,75 +525,109 @@ export class OkexPerpetualSwapStrategy implements HedgingStrategy {
...
+                  { withdrawalResponse },
+                  "rebalancing withdrawal was NOT successful",
+                )
+              }
             } else {
-              await database.externalTransfers.insert(extTransferRecord)
-              this.logger.error(
-                { withdrawalResponse },
-                "rebalancing withdrawal was NOT successful",
+              this.logger.debug(
```
Comment (dolcalmi) [inline dealer/src/OkexPerpetualSwapStrategy.ts:623]: same issue here with the else block.. code is cleaner using an early return

**Review by dolcalmi — APPROVED** [review]
Approving but with some comments. Not sure if worth to do a lot of changes to code style if we will re-write the dealer

File: dealer/src/OkexPerpetualSwapStrategy.ts (hunk 1 above)
Comment (sebastienverreault) [reply dealer/src/OkexPerpetualSwapStrategy.ts:544]: Yeah if it fails too bad or should it be an alert to get attention?

**Pull request merged**